pkg/querier/tenantfederation/merge_queryable_test.go

```diff
@@ -195,241 +213,465 @@ func (m mockTenantQuerier) LabelNames() ([]string, storage.Warnings, error) {
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a", "host2.team-b", "host2.team-c"},
+				},
+				{
+					name:                "should return all tenant team values for the __tenant_id__ label when no matchers are provided",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: []string{"team-a", "team-b", "team-c"},
+				},
+				// TODO(jdb): These tests are failing as matchers are not implemented. They should pass when the matchers are implemented
+				// 	{
+				// 		name:                "should return only label values for team-a and team-c tenants when there is a not-equals matcher for team-b tenant",
+				// 		labelName:           defaultTenantLabel,
+				// 		matchers:            []*labels.Matcher{{Name: defaultTenantLabel, Value: "team-b", Type: labels.MatchNotEqual}},
+				// 		expectedLabelValues: []string{"team-a", "team-c"},
+				// 	},
+				// 	{
+				// 		name:                "should return only label values for team-b tenant when there is an equals matcher for team-b tenant",
+				// 		labelName:           defaultTenantLabel,
+				// 		matchers:            []*labels.Matcher{{Name: defaultTenantLabel, Value: "team-b", Type: labels.MatchEqual}},
+				// 		expectedLabelValues: []string{"team-b"},
+				// 	},
+			},
+		},
+		{
+			mergeQueryableScenario: threeTenantsWithDefaultTenantIDScenario,
+			labelValuesTestCases: []labelValuesTestCase{
+				{
+					name:                "should return all label values for instance when no matchers are provided",
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a", "host2.team-b", "host2.team-c"},
+				},
+				{
+					name:                "should return all tenant values for __tenant_id__ label name",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: []string{"team-a", "team-b", "team-c"},
+				},
+				{
+					name:                "should return the original value for the revised tenant labelname when no matchers are provided",
+					labelName:           originalDefaultTenantLabel,
+					expectedLabelValues: []string{"original-value"},
+				},
+				{
+					name:             "with equals matcher for __original_tenant_id__ label it should error as matchers aren't implemented for LabelValues yet",
+					matchers:         []*labels.Matcher{{Name: originalDefaultTenantLabel, Value: "original-value", Type: labels.MatchEqual}},
+					labelName:        originalDefaultTenantLabel,
+					expectedQueryErr: errMatchersNotImplemented,
+				},
+			},
+		},
+		// TODO(jdb): Should these errors be warnings and errors be present?
+		// Currently they are not returned by the LabelValues.
+		// {
+		// 	mergeQueryableScenario: threeTenantsWithWarningsScenario,
+		// 	labelValuesTestCases: []labelValuesTestCase{{
+		// 		name:                "warnings from all merged queryables should be returned",
+		// 		labelName:           defaultTenantLabel,
```

Comment:
Queries with LabelName defaultTenantLabel never reach the underlying queries, so this is expected:

https://github.com/cortexproject/cortex/blob/fde0a6276446bcd806d82209f9d7d6e3f70eec83/pkg/querier/tenantfederation/merge_queryable.go#L136